### PR TITLE
Feature/set up travis

### LINKED ISSUE #6: Set up TravisCI with deploy to Heroku

### PR DESCRIPTION
**Pull request by knees4bees**
### What does this PR do?
* Adds continuous integration with Travis CI and auto-deployment to Heroku (which may or may not be working -- can't confirm until we have something new on `main` because the auto-deploy is only triggered from `main`).
* Fixes tests -- they are all passing now.

### Anything specific the reviewer should look at?
* The config is in `package.json` and `.travis.yml`. The test changes are mostly updates to `cy.visit` commands in `student-test.js`.

Closes #6

### CODE REVIEW

**Review by m1073496 — COMMENTED** [review]
Thanks for tackling this Katie, you're a trooper! I'm going to leave the merge for the group, since I'm not totally sure about what a merge will do with the CI stuff.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
-  # - "npm run ci"
+- node
+script:
+- npm run ci
 notifications:
   email: false
+deploy:
+  provider: heroku
+  api_key:
+    secure: fVyQxfoGuJXaQ67fWoIPNbOLoF9cVHhIs6w+lvOQUEfeyG6ICEy21LTTQcirZ0HjE7n3Ysxqg2po3+o8B00UMTtz4XgONCny6Czoxee1+IE1xDXyxteHroJNb5E08QNZ7gMgr5aBrNVZXGDRGiw7Wv/Xs8JVEqsve4gQL/bvAMS9HugEN5QttPvxtPAslXRtudkHOgzq/APqug/CemcoOUPdW1BJvo6nKDOQ17NFi5qq1Hybr3m0lGHpKGPoIFZPyZG2+0NXUGkS+3Ebonv2PC7DieShIfb6xuhH1Pb3hM6UJb4qu491n/KD4KfbCszR46hAH+bvJsgsp6cQKEFKsC8TtO89VK2WJArH/vZvR96CKnDVNQi0/rzpbLJgNq7SA+k4DdvTpiKqvJ4GePm1k1NaEEmEUXEjATCRJy9N7Xqpl4mBUceGGc7+H26g+XxohzFTsHHCni4C/KlBH6i4QgcVXUHkLdwmLTfaFV8XRf0fzXgiE3i0C2XecjvWIH6NMTKE2nux/V8lP3jiTCah3y80H/qFARywbSUzg6p5QzZEeoLiJigqa9FF8cIj0VsKD/iu8IizCSwa1q35VD8Z6+vcWLqds39BR8NGWsZyEvWfTCHy+ULoWG8EjM2l/FP62+/Ym+WJWjv3BllKyTvRAmPijy0rU0q9cAkFVjuYmOk=
```
Comment (piknikki) [inline .travis.yml:11]: ohhhhhh I did not realize you didn't need quotes. I totally missed that.

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
     "test": "vue-cli-service test:e2e",
+    "ci": "start-server-and-test serve http://localhost:8080 cy-run",
```
Comment (piknikki) [inline package.json:9]: Great job working through this!

File: tests/e2e/specs/student-test.js [hunk 3]
```diff
@@ -11,14 +17,14 @@ describe('Mesh landing page after login', () => {
...
 })
 
 // this one isn't working for some reason. I think I need to change the route or test the route by clicking???
 describe('Mesh about page', () => {
-  it.skip('Visits the about route and renders without errors', () => {
-    cy.visit('/about')
-    cy.get('.feedback-container').children('.about').should('exist')
+  it('Visits the about route and renders without errors', () => {
+    cy.visit('http://localhost:8080/about')
+    cy.get('h1').should('exist')
```
Comment (piknikki) [inline tests/e2e/specs/student-test.js:28]: We can start writing real tests as soon as our UI has all the kinks worked out. Hopefully today-tomorrow-ish. Thank you so much for your hard work on TravisCI!

**Review by piknikki — COMMENTED** [review]
Thanks for all your work on Travis. Your diligence is awesome.

**Pull request merged**